### PR TITLE
:seedling: Add PR verifier

### PR DESCRIPTION
**Pull request by DirectXMan12**
This adds a PR description/title verifier using
[kubebuilder-release-tools](https://sigs.k8s.io/kubebuilder-release-tools).
It checks that PRs have the correct type indicator emoji, and does some
basic descriptiveness checks.

This also updates the PR template to mention the emoji, and adds a VERSIONING.md file describing the processes/linking to the common processes..

### CODE REVIEW

**Review by vincepri — APPROVED** [review]
/lgtm

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Comment by DirectXMan12** [comment]
(by the power vested in me, manually re-applying LGTM since the update was just to generated files that should've been updated previously but were not)

File: VERSIONING.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+[envtest-ref]: https://book.kubebuilder.io/reference/artifacts.html
+[tools-branch]: https://github.com/kubernetes-sigs/kubebuilder/tree/tools-releases)
+
+## Understanding the versions
+
+|   Name	|   Example	|  Description |
+|---	|---	|---	|
+|  PROJECT version |  `v1`,`v2`,`v3` | As of the introduction of [Extensible CLI and Scaffolding Plugins](https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/extensible-cli-and-scaffolding-plugins-phase-1.md), PROJECT version represents the layout of PROJECT file itself.  For `v1` and `v2` projects, there's extra meaning -- see below.  |
+|  Release version | `v2.2.0`, `v2.3.0`, `v2.3.1` | Tagged versions of the KubeBuilder project, representing changes to the source code in this repository. See the [releases](https://github.com/kubernetes-sigs/kubebuilder/releases) page. |
```
Comment (camilamacedo86) [inline VERSIONING.md:67]: We need to change these content, however, it is not part of the scope of this Pr so. 👍

**Comment by DirectXMan12** [comment]
/retest

kind flake

**Comment by DirectXMan12** [comment]
next up: update the golangci-lint action to use problem matchers or the checks framework directly

**Pull request merged**